### PR TITLE
TILA-727 | Add gql query to query reservation units by unit

### PR DESCRIPTION
**Pull request by nicobav**
This is now basing the TILA-726 branch for diff's sake. When that gets merged can change the base to main.

### CODE REVIEW

File: api/graphql/schema.py [hunk 1]
```diff
@@ -37,9 +43,17 @@ class AllowAuthenticatedFilter(AuthFilter):
     permission_classes = (AllowAuthenticated,)
 
 
-class ReservationUnitsFilter(AuthFilter):
+class ReservationUnitsFilter(AuthFilter, django_filters.FilterSet):
     permission_classes = (ReservationUnitPermission,)
 
+    unity = ModelInFilter(field_name="unit", queryset=Unit.objects.all())
```
Comment (aprenfors) [inline api/graphql/schema.py:49]: Unity?

Comment (nicobav) [reply api/graphql/schema.py:49]: :see_no_evil:  typo => i'll fix

Comment (nicobav) [reply api/graphql/schema.py:49]: oh, actually this is completely unnecessary here => was testing FilterSets with bad results => I'll remove this `FilterSet` experiment here

Comment (nicobav) [reply api/graphql/schema.py]: Okay, actually got the `FilterSet` class to work.

**Comment by nicobav** [comment]
Added new `FilterSet` class for `ReservationUnitFIlter` => naming the `FilterSet` class as `ReservationUnitFilterSet` (usually naming convention is just `NNNFilter` but it would be confusing with the django filter connections :exploding_head: 
Anyhow the unit filtering works now nicely within the `reservationUnits` filter

**Review by aprenfors — APPROVED** [review]

**Pull request merged**